iati/core/tests/test_rulesets.py

```diff
@@ -222,54 +232,66 @@ def rule_constructor(self, rule_type):
             Differs from fixture of same name in TestRuleSubclass as specifies rule_type via specific TestRuleSubclass fixture.
 
         """
-        return iati.core.rulesets.locate_constructor_for_rule_type(rule_type)
+        return iati.core.rulesets.constructor_for_rule_type(rule_type)
+
+    @pytest.fixture
+    def valid_condition_rule(self, rule_constructor, valid_single_context, valid_condition_case):
+        """Return a Rule with a `condition` case."""
+        return rule_constructor(valid_single_context, valid_condition_case)
+
+    @pytest.fixture
+    def invalid_condition_rule(self, rule_constructor, valid_single_context, invalid_condition_case):
+        """Return a Rule with a `condition` case."""
+        return rule_constructor(valid_single_context, invalid_condition_case)
 
     def test_rule_init_valid_parameter_types(self, rule_basic_init):
         """Check that Rule subclasses can be instantiated with valid parameter types."""
         assert isinstance(rule_basic_init, iati.core.Rule)
 
-    def test_rule_init_valid_empty_path_case(self, rule_valid_empty_case):
-        """Check that Rule subclasses can be instantiated with a valid empty case parameter."""
-        assert isinstance(rule_valid_empty_case, iati.core.Rule)
+    def test_rule_init_raises_error_on_empty_context(self, rule_constructor, instantiating_case):
+        """Check that a Rule cannot be instantiated when the `context` is an empty string."""
+        invalid_context = ''
+        with pytest.raises(ValueError):
+            rule_constructor(invalid_context, instantiating_case)
 
     def test_rule_attributes_from_case(self, rule_basic_init):
         """Check that a Rule subclass has case attributes set."""
-        required_attributes = rule_basic_init._required_case_attributes(rule_basic_init._ruleset_schema_section())
+        required_attributes = rule_basic_init._case_attributes(rule_basic_init._ruleset_schema_section())
         for attrib in required_attributes:
             # Ensure that the attribute exists - if not, an AttributeError will be raised
             getattr(rule_basic_init, attrib)
 
+    def test_optional_rule_attributes_from_case(self, rule_constructor, valid_single_context, valid_condition_case):
+        """Check that a Rule subclass has case attributes set."""
```

Comment:
Should specify in the docstring that are looking for optional rather than mandatory attributes.

Reply:
done